ai-decision-services/ai-ad-dpp-engine/prepare-config-data-and-app/prepare-config-data-and-app.md

```diff
@@ -0,0 +1,217 @@
+    *   **user** \- the owner of the table. In most cases it will be **admin**. 
+    *   **password** \- the password you setup when you download the Wallet. 
+
+## phaseInfo
+
+The flag indicating whether it is a training or inferencing phase, plus other required input parameters. Here we will include an object storage connector, the workflow will use it to store or read metadata, which is useful for data dependent steps (for example, one-hot encoding).
+
+*   **connector** - the connector to object storage. It will be composed of
+    *   **namespace** - the namespace of the target object storage bucket. Required if it is for object storage.
+    *   **bucket** - the bucket name. Required if it is for object storage.
+    *   **objectName** \- the full path and name of the object. Required if it is for object storage.
+
+## processingSteps
+
+This will be a list of all the transformations you want to apply. This is the core of the data processing. Every item in the list is a step, and all the steps will be executed in sequence, one after another. Users should make sure the sequence makes sense, or exceptions will be thrown. DPP doesn't responsible for process validation before execution. For each step, you need to specify the following.
+
+*   **stepType** \- can be either **singleDataFrameProcessing** (for transforming on top of single dataframe) or **combineDataFrames** (for merging or joining two dataframes).
+*   **configurations** - the configuration related to the step you take, which will include:
+    *   **dataframeName** - a name you given to the output of the transformation.
+    *   **steps** - a list of all transformers you want to apply to your target dataframe at this time. For each step, there will be:
+        *   **stepName** - the name of the transformer you want to apply. It must exactly matches the one you defined in archive. For the full list of offered transformers, please check **Appendix II: List of Transformers**.
+        *   **args** \- all the arguments required for the transformer you choose to use. For details about it, check **Appendix II: List of Transformers**.
+
+## stagingDestination
+
+This is where we store intermediate processed data for post processing. Basically these will be dataframes stored in csv format at an assigned object storage bucket. 
+
+*   **combinedResult (optional)** - the final dataframe you want to output. You need to specify this parameter if you only have **one** single dataframe as the output. But if you do sharding, the sharded dataset will be the output automatically, and you don't need this parameter any more.
+*   **type** \- can only be **object-storage** for now.
+*   **namespace** - the namespace of the target object storage bucket. 
+*   **bucket** \- the bucket name. 
+*   **objectName** \- the full path and name of the object. 
+
+## outputDestination
+
+This is where the final output comes to, including model\_info, finalized output after inferencing, etc.
```

Comment:
Reword.  Should this be - This is where the final output (AD results) is stored after inferencing (?)